vulture_os/toolkit/api_parser/sentinel_one_identity_graph/sentinel_one_identity_graph.py

```diff
@@ -0,0 +1,263 @@
+                    detectedAt
+                    detectionSource {...DetectionSource}
+                    detectionTime {...DetectionTime}
+                    externalId
+                    fileHash
+                    fileName
+                    firstSeenAt
+                    id
+                    indicators {...Indicators}
+                    lastSeenAt
+                    name
+                    noteExists
+                    process {...Process}
+                    realTime {...RealTime}
+                    result
+                    severity
+                    status
+                    storylineId
+                    updatedAt
+                }
+            }
+        """
+
+        response = self.execute_query(
+            self.url,
+            data={'query': query, 'variables': {'id': event_id}}
+        )
+
+        return response.get('data', {}).get('alert', {})
+
+    def format_log(self, log: dict) -> str:
+        return dumps(log)
+
+    def test(self):
+        logs = []
+
+        try:
+            since = timezone.now() - timedelta(days=1)
+            to = timezone.now()
+
+            alert_ids = self.get_itdr_events_ids(
+                since=int(since.timestamp()*1000),
+                to=int(to.timestamp()*1000),
+                limit=100)
+
+            for id in alert_ids:
+                logs.append(self.get_itdr_event_details(id))
+
+            return {
+                "status": True,
+                "data": logs
+            }
+        except Exception as e:
+            return {
+                "status": False,
+                "error": str(e)
+            }
+
+    def execute(self):
+        kind = "sentinel_one_identity_graph_alerts"
+
+        current_time = timezone.now()
+
+        since = datetime.strptime(self.last_collected_timestamps.get(kind), "%Y-%m-%dT%H:%M:%SZ") or current_time - timedelta(days=7)
+        to = min(since + timedelta(days=1), current_time - timedelta(minutes=3))
+
+        while to < current_time - timedelta(minutes=3):
+            since = datetime.strptime(self.last_collected_timestamps.get(kind), "%Y-%m-%dT%H:%M:%SZ") or timezone.now() - timedelta(days=7)
+            to = min(since + timedelta(days=1), current_time - timedelta(minutes=3))
+
+            alert_ids = self.get_itdr_events_ids(
+                since=int(since.timestamp()*1000),
+                to=int(to.timestamp()*1000),
+                limit=100
+            )
+
+            logger.info(f"[{__parser__}][execute]: Successfully gets {len(alert_ids)} alert's ids", extra={'frontend': str(self.frontend)})
+
+            for id in alert_ids:
+                event_detail = self.get_itdr_event_details(id)
+                self.write_to_file(event_detail)
+
+            if len(alert_ids) > 0 or since < current_time - timedelta(hours=24): # if logs or timestamp to old
+                self.last_collected_timestamps[kind] = datetime.strftime(to, "%Y-%m-%dT%H:%M:%SZ")
```

Comment:
objects in `last_collected_timestamps` should always be (timezone-aware) datetime objects